src/utils.jl

```diff
@@ -47,3 +49,11 @@ function _is_select_function_trial(f, ::Type{T1}, ::Type{T2}) where {T1,T2}
         return false
     end
 end
+
+saturating_add(x::X, y::X) where {X<:FixedPoint} =
```

Comment:
style: People commonly use `x::T` for types, `img::AT` for array types, and `c::CT` for Colorant Types.